plugins/mattermost/server/api/mattermost.ts

```diff
@@ -0,0 +1,160 @@
+import Router from "koa-router";
+import { IntegrationService, IntegrationType, UserRole } from "@shared/types";
+import { MattermostIntegrationSettings } from "@shared/zod";
+import auth from "@server/middlewares/authentication";
+import { transaction } from "@server/middlewares/transaction";
+import validate from "@server/middlewares/validate";
+import { Collection, Event, Integration } from "@server/models";
+import { authorize } from "@server/policies";
+import { presentIntegration, presentPolicies } from "@server/presenters";
+import { APIContext } from "@server/types";
+import {
+  createWebhook,
+  getChannels,
+  getUser,
+  getUserTeams,
+} from "../mattermost/client";
+import { loadChannelsFromCache, storeChannelsInCache } from "../utils/cache";
+import * as T from "./schema";
+
+const router = new Router();
+
+router.post(
+  "mattermost.user_teams",
+  auth({ role: UserRole.Admin }),
+  validate(T.MattermostGetUserTeamsSchema),
+  async (ctx: APIContext<T.MattermostGetUserTeamsReq>) => {
+    const { url, apiKey } = ctx.input.body;
+
+    const [user, teams] = await Promise.all([
+      getUser({ serverUrl: url, apiKey }),
+      getUserTeams({ serverUrl: url, apiKey }),
+    ]);
+
+    ctx.body = {
+      data: { user, teams },
+    };
+  }
+);
+
+router.post(
+  "mattermost.channels",
+  auth({ role: UserRole.Admin }),
+  validate(T.MattermostGetChannelsSchema),
+  transaction(),
+  async (ctx: APIContext<T.MattermostGetChannelsReq>) => {
+    const { user } = ctx.state.auth;
+    const { force } = ctx.input.body;
+
+    const integration = await Integration.scope("withAuthentication").findOne({
+      where: {
+        service: IntegrationService.Mattermost,
+        type: IntegrationType.LinkedAccount,
+        userId: user.id,
```

Comment:
```suggestion
        teamId: user.teamId,
```

This should probably be scoped to the team rather than the specific user?